Bottom: b857c6917ad1e7f3e08d704b2880a2a697545152
Top:    b0708bc191b67666365f29d9c0698c15e9728bbb
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-03 20:14:36 +0000

Refresh of add-the-bloat-as-a-parameter

---

diff --git a/src/find.c b/src/find.c
index 05081b0..bff8717 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3321,6 +3321,17 @@ struct drc_info
   int flag;
 };
 
+static void
+start_do_it_and_dump (int type, void *ptr1, void *ptr2, void *ptr3,
+                      int flag, bool AndDraw,
+                      Coord bloat)
+{
+  Bloat = bloat;
+  ListStart (type, ptr1, ptr2, ptr3, flag);
+  DoIt (flag, true, AndDraw);
+  DumpList ();
+}
+
 /*-----------------------------------------------------------------------------
  * Check for DRC violations on a single net starting from the pad or pin
  * sees if the connectivity changes when everything is bloated, or shrunk
@@ -3337,11 +3348,8 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
 
   if (PCB->Shrink != 0)
     {
-      Bloat = -PCB->Shrink;
-      ListStart (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG);
-      DoIt (DRCFLAG | SELECTEDFLAG, true, false);
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG, false, -PCB->Shrink);
       /* ok now the shrunk net has the SELECTEDFLAG set */
-      DumpList ();
       ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
       Bloat = 0;
       drc = true;               /* abort the search if we find anything not already found */
@@ -3352,15 +3360,9 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
           ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
           User = true;
           drc = false;
-          Bloat = -PCB->Shrink;
-          ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
-          DoIt (SELECTEDFLAG, true, true);
-          DumpList ();
-          ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
-          Bloat = 0;
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, -PCB->Shrink);
           drc = true;
-          DoIt (FOUNDFLAG, true, true);
-          DumpList ();
+          start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, 0);
           User = false;
           drc = false;
           drcerr_count++;
@@ -3392,10 +3394,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
   /* now check the bloated condition */
   drc = false;
   ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
-  Bloat = 0;
-  ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
-  DoIt (SELECTEDFLAG, true, false);
-  DumpList ();
+  start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, false, 0);
   flag = FOUNDFLAG;
   ListStart (What, ptr1, ptr2, ptr3, flag);
   Bloat = PCB->Bloat;
@@ -3407,15 +3406,9 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
       User = true;
       drc = false;
-      Bloat = 0;
-      ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
-      DoIt (SELECTEDFLAG, true, true);
-      DumpList ();
-      ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
-      Bloat = PCB->Bloat;
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, SELECTEDFLAG, true, 0);
       drc = true;
-      DoIt (FOUNDFLAG, true, true);
-      DumpList ();
+      start_do_it_and_dump (What, ptr1, ptr2, ptr3, FOUNDFLAG, true, PCB->Bloat);
       drcerr_count++;
       LocateError (&x, &y);
       BuildObjectList (&object_count, &object_id_list, &object_type_list);
@@ -3442,10 +3435,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       Undo (true);
       /* highlight the rest of the encroaching net so it's not reported again */
       flag = FOUNDFLAG | SELECTEDFLAG;
-      Bloat = 0;
-      ListStart (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag);
-      DoIt (flag, true, true);
-      DumpList ();
+      start_do_it_and_dump (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag, true, 0);
       drc = true;
       Bloat = PCB->Bloat;
       ListStart (What, ptr1, ptr2, ptr3, flag);
